Remove `window.data` usage.

davidlehn opened the pull request:
- I'm unsure if that component `$config` code is used anywhere else.
- Should it default to `{}` since `window.data` was removed?

gannan08 commented:
Is moving the backend configured config into the frontend an option? This way we continue to have the same variable attached to the window and we override it for different frontend applications.

davidlehn committed:
Remove `window.data` usage, use `supportEmail`.

- No longer use `window.data`.
- Use `supportEmail` config vs `supportDomain`.

davidlehn committed:
Use a common global config schema.

aljones15 commented:
has this been tested with an actual not found error? Also will set the support email for [...] today as me and @gannan08 were discussing one recently.

davidlehn committed:
Add eslint support.

davidlehn committed:
Fix lint issues.

davidlehn committed:
Move config to config.js.

davidlehn committed:
Handle HTTP meta error codes.

- Add errors to config.
- Add error pages.
- Add custom base error page component name to config.
- Use custom endpoints to display error when found on initial load.
- Use common `bedrock-vue-core` for most dynamicly loaded chunks.

davidlehn committed:
Update error handling.

- Remove HTTP error handling. HTML meta HTTP status code support removed
  from bedrock-views.
- Update changelog.
- Minor fixes.
